### PR TITLE
Watch JS files in development docker compose

### PR DESCRIPTION
**Pull request by nllong**
This functionality does slow down the initial load time of the docker-compose.dev.yml file due to having to webpack the contents after the server starts. Any touch of a JS file will trigger a rebuild, which could probably be sped up.

* add nodemon
* point worker_web multistage to dev
* Do not copy poetry lock and package-lock to containers
* when mounting web files, ignore build and node_modules

How to test:
* Launch the docker containers with `docker compose -f docker-compose.yml -f docker-compose.dev.yml up --build`
* run integration tests `poetry run pytest -m "integration"`
* touch one of the server JS files and make sure the site rebuilds
* monkey test the site as needed

### CODE REVIEW

File: alfalfa_web/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "clean": "npx rimraf build",
     "start": "node ./build/index.js",
     "start-dev": "npm run build-server && env-cmd -f ../.env npm start",
-    "watch": "webpack --watch --progress"
+    "watch": "nodemon -e js,html --ignore build/ --exec \"npm run build-app && npm run build-server && npm start\""
```
Comment (nllong) [inline alfalfa_web/package.json:13]: this is the main change to make the watch code work. I couldn't see a way to build the server and the app with just webpack. Maybe I missed something.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -47,27 +50,22 @@ jobs:
...
         uses: actions/checkout@v2
 
+      # only need to start the worker which saves time here because
+      # the webpack calling on the web container for .dev.yml can take some
+      # time.
       - name: Build and run stack
         run: |
           printenv
-          docker-compose -f docker-compose.yml -f docker-compose.dev.yml up --build -d
+          docker-compose -f docker-compose.yml -f docker-compose.dev.yml up --build -d worker
```
Comment (TShapinsky) [inline .github/workflows/ci.yml:66]: Is this just to try to speed up tests?

Comment (nllong) [reply .github/workflows/ci.yml:66]: yup

**Comment by nllong** [comment]
@axelstudios  and @TShapinsky  - this is ready for another review. I updated a few items based on Alex's feedback (e.g., updating energyplus direction, removing the unused .dockerignore in alfafla_web, keeping poetry.lock and package-lock.json).

**Review by TShapinsky — APPROVED** [review]
This seems fine to me

**Comment by nllong** [comment]
@axelstudios -- can you check real quick? I'd like to rebase my `add-mongoengine` against this before sending to Toby to debug... thanks!

**Review by axelstudios — APPROVED** [review]

**Pull request merged**